src/components/AvatarWithImagePicker.js

```diff
@@ -116,112 +114,115 @@ const defaultProps = {
+    anchorPosition, 
+    anchorAlignment, 
+    onImageSelected, 
+    editorMaskImage,
+}) {
+
+    const [isMenuVisible, setIsMenuVisible] = useState(false);
+    const [errorData, setErrorData] = useState({
+        validationError: null,
+        phraseParam: {},
+    });    
+    const [isAvatarCropModalOpen, setIsAvatarCropModalOpen] = useState(false);
+    const [imageData, setImageData] = useState({
+        uri: '',
+        name: '',
+        type: ''
+    });
+    const anchorRef = useRef();
+    const {translate} = useLocalize();
+
+    useEffect(() => {
+        // check if the component is no longer focused, then reset the error.
+        if (!isFocused) {
+            setError(null, {});
         }
-
-        this.animation.start();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isFocused && this.props.isFocused) {
-            this.setError(null, {});
-        }
-        if (!prevProps.isUploading && this.props.isUploading) {
-            this.animation.start();
-        } else if (prevProps.isUploading && !this.props.isUploading) {
-            this.animation.stop();
-        }
-    }
-
-    componentWillUnmount() {
-        this.animation.stop();
-    }
+    }, [isFocused]);
 
     /**
      * @param {String} error
      * @param {Object} phraseParam
      */
-    setError(error, phraseParam) {
-        this.setState({validationError: error, phraseParam});
-    }
+    const setError = (error, phraseParam) => {
+        setErrorData({
+            validationError: error,
+            phraseParam: phraseParam,
+        });
+    };
 
     /**
      * Check if the attachment extension is allowed.
      *
      * @param {Object} image
      * @returns {Boolean}
      */
-    isValidExtension(image) {
+    const isValidExtension = (image) => {
         const {fileExtension} = FileUtils.splitExtensionFromFileName(lodashGet(image, 'name', ''));
         return _.contains(CONST.AVATAR_ALLOWED_EXTENSIONS, fileExtension.toLowerCase());
-    }
+    };
 
-    /**
+     /**
      * Check if the attachment size is less than allowed size.
      *
      * @param {Object} image
      * @returns {Boolean}
      */
-    isValidSize(image) {
+    const isValidSize = (image) => {
         return image && lodashGet(image, 'size', 0) < CONST.AVATAR_MAX_ATTACHMENT_SIZE;
-    }
+    };
 
     /**
      * Check if the attachment resolution matches constraints.
      *
      * @param {Object} image
      * @returns {Promise}
      */
-    isValidResolution(image) {
-        return getImageResolution(image).then(
-            (resolution) =>
-                resolution.height >= CONST.AVATAR_MIN_HEIGHT_PX &&
-                resolution.width >= CONST.AVATAR_MIN_WIDTH_PX &&
-                resolution.height <= CONST.AVATAR_MAX_HEIGHT_PX &&
-                resolution.width <= CONST.AVATAR_MAX_WIDTH_PX,
-        );
-    }
+    const isValidResolution = async (image) => {
+        const resolution = await getImageResolution(image);
```

Comment:
https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#esnext-are-we-allowed-to-use-insert-new-language-feature-why-or-why-not

> Async/Await - Use the native Promise instead

<img width="723" alt="image" src="https://github.com/Expensify/App/assets/9059945/a5f2ecda-bebc-4acd-9906-0491e744d470">